Documentation for LocalQueueDefaulting (#3783)
kubernetes-sigs/kueue: site/content/en/docs/tasks/manage/enforce_job_management/setup_default_local_queue.md

---
title: "Setup default LocalQueue"
date: 2024-12-12
weight: 10
description: >
  Setup default LocalQueue to fullfil a queue label on jobs that submited without queue label.
---

> This page describes how to setup default LocalQueue to ensure that all Workloads submitted to specific namespace
> intended for use by _batch users_ will be managed by Kueue even if they lack a `kueue.x-k8s.io/queue-name` label.

Reviewer: ```suggestion
This page describes how to setup default LocalQueue to ensure that all Workloads submitted to a specific namespace, intended for Kueue users, are managed by Kueue, even if the `kueue.x-k8s.io/queue-name` label is not specified explicitly. 
```

## Setup default LocalQueue

LocalQueueDefaulting is an Alpha feature that allows the use of a LocalQueue with name `default` as the default LocalQueue for workloads in the same namespace that do not have the `kueue.x-k8s.io/queue-name` label. The feature is gated by the `LocalQueueDefaulting` feature gate, and is disabled by default. To use this feature:

> - Enable the LocalQueueDefaulting feature gate. Refer to the [feature gates configuration](/docs/installation/#change-the-feature-gates-configuration) guide for details.
> - create a LocalQueue with the name `default`.
> - create a workload in the same namespace. Observe that workload is updated with the `kueue.x-k8s.io/queue-name:default` label.
> - workloads created in a different namespace or workloads that already have the `kueue.x-k8s.io/queue-name` label won't be modified.

Reviewer: ```suggestion
- Enable the LocalQueueDefaulting feature gate. Refer to the [feature gates configuration](/docs/installation/#change-the-feature-gates-configuration) guide for details.
- create a LocalQueue with the name `default` in a namespace. 

That's all! Now, to test the feature, create a Job in the same namespace. Observe that the Job is updated with the `kueue.x-k8s.io/queue-name: default` label.

Note that workloads created in a different namespace or workloads that already have the `kueue.x-k8s.io/queue-name` label won't be modified.
```

Reviewer: nit proposal. As the 2 last points aren't really the setup, but testing and a note

Author: Thank you!

---
title: "Setup default LocalQueue"
date: 2024-12-12
weight: 10
description: >
  Setup default LocalQueue to fullfil a queue label on jobs that submited without queue label.
---

> This page describes how to setup default LocalQueue to ensure that all Workloads submitted to a specific namespace, intended for Kueue users, are managed by Kueue, even if the `kueue.x-k8s.io/queue-name` label is not specified explicitly. 

Reviewer: ```suggestion
This page describes how to setup default LocalQueue to ensure that all Workloads submitted to a specific namespace are managed by Kueue, even if the `kueue.x-k8s.io/queue-name` label is not specified explicitly. 
```
maybe it is not that relevent that it is "indended for Kueue users". And dropping this part makes it more readable imo.

## Setup default LocalQueue

LocalQueueDefaulting is an Alpha feature that allows the use of a LocalQueue with name `default` as the default LocalQueue for workloads in the same namespace that do not have the `kueue.x-k8s.io/queue-name` label. The feature is gated by the `LocalQueueDefaulting` feature gate, and is disabled by default. To use this feature:

> - Enable the LocalQueueDefaulting feature gate. Refer to the [feature gates configuration](/docs/installation/#change-the-feature-gates-configuration) guide for details.
> - Enable the LocalQueueDefaulting feature gate. Refer to the [feature gates configuration](/docs/installation/#change-the-feature-gates-configuration) guide for details.

Reviewer: ```suggestion
- Enable the LocalQueueDefaulting feature gate. Refer to the [feature gates configuration](/docs/installation/#change-the-feature-gates-configuration) guide for details.
```
it is duplicated

---
title: "Setup default LocalQueue"
date: 2024-12-12
weight: 10
description: >
  Setup default LocalQueue to fullfil a queue label on jobs that submited without queue label.
---

This page describes how to setup default LocalQueue to ensure that all Workloads submitted to a specific namespace are managed by Kueue,
even if the `kueue.x-k8s.io/queue-name` label is not specified explicitly.

## Setup default LocalQueue

> LocalQueueDefaulting is an Alpha feature that allows the use of a LocalQueue with name `default` as the default LocalQueue

Reviewer: I would prefer to add the feature banner as here: https://github.com/kubernetes-sigs/kueue/blob/e544dc8a447abfafb471886f9390af0a303b24d5/site/content/en/docs/concepts/topology_aware_scheduling.md?plain=1#L9. Can be a follow up.

Reviewer 2: This was addressed in the https://github.com/kubernetes-sigs/kueue/pull/3867
